modules/modules.mk

```diff
@@ -180,6 +181,13 @@ ifeq ($(XFEM),yes)
   include $(FRAMEWORK_DIR)/app.mk
 endif
 
+ifeq ($(EXTERNAL_PETSC_SOLVER),yes)
+  APPLICATION_DIR    := $(MOOSE_DIR)/modules/external_petsc_solver
+  APPLICATION_NAME   := external_petsc_solver
+
```

Comment:
Oh you need a suffix - that's super important. In fact I'm surprised any of this works without it!